pkg/endpoint/endpoint.go

```diff
@@ -277,16 +249,56 @@ func (e *EndpointStatus) lastIndex() int {
 	return lastIndex
 }
 
+// getAndIncIdx returns current free slot index and increments the index to the
+// next index that can be overwritten.
 func (e *EndpointStatus) getAndIncIdx() int {
 	idx := e.Index
 	e.Index++
 	if e.Index >= maxLogs {
 		e.Index = 0
 	}
+	// Lets skip the CurrentStatus message from the log to prevent removing
+	// non-OK status!
+	if e.Index < len(e.Log) &&
+		e.Log[e.Index] == e.CurrentStatus &&
+		e.CurrentStatus.Status.Code != OK {
+		e.Index++
+		if e.Index >= maxLogs {
+			e.Index = 0
+		}
+	}
 	return idx
 }
 
-func (e *EndpointStatus) addStatusLog(s *statusLog) {
+// addStatusLog adds statusLogMsg to endpoint log.
```

Comment:
I'm not sure I fully understand the concept yet. The fundamental problem is that you have multiple causes for an endpoint to be considered in an failed state:
 - bpf generation or compilation
 - policy calculation
 - ....

Each of these causes can be OK/warning/failure. You want to report an overall status. You also want to store a log of status changes for each. The priority seems to indicate that a BPF failure always overwrites a policy failure but that seems incomplete.

Let's assume:
1. bpf fails
2. policy fails
3. bpf succeeds

You still want the overall state to say failure.

I think instead of priority, what you want is this:
```map[StatusComponent]statusLogMsg```

BPF and priority are both `StatusComponent`

OverallStatus is Failure if any of the components failed, else warning if any component is warning, else OK.

The log shows the latest n status changes of the endpoint.